### ISSUE TITLE
Weird Zoom issue on Electron Apps (native ozone only)

### ISSUE THREAD

**Issue by cpiber** [hyprwm/Hyprland#8055]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit bc299928ad5571300180eb8edb6742ed3bbf0282  ().
Date: Wed Oct 9 01:26:40 2024
Tag: v0.44.0-20-gbc299928, commits: 5319
built against aquamarine 0.4.2


flags: (if any)


System Information:
System name: Linux
Node name: PiW22p
Release: 6.8.0-45-generic
Version: #45-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 30 12:02:04 UTC 2024


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-P GT2 [Iris Xe Graphics] [8086:46a6] (rev 0c) (prog-if 00 [VGA controller])
2f:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP104 [GeForce GTX 1070] [10de:1b81] (rev a1) (prog-if 00 [VGA controller])
NVRM version: NVIDIA UNIX x86_64 Kernel Module  560.35.03  Fri Aug 16 21:39:15 UTC 2024


os-release: PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo


plugins:
  hyprscroller by dawser ver 1.0

======Config-Start======


Config File: /home/constantin/.config/hypr/hyprland.conf: Read Succeeded
#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

# See https://wiki.hyprland.org/Configuring/Monitors/
monitor=eDP-1,preferred,auto,1.5
#monitor=eDP-1,disable
monitor=HDMI-A-1,preferred,auto-left,1
# monitor=HDMI-A-1,preferred,auto-right,1
monitor=HDMI-A-2,preferred,auto-left,1
monitor=DP-4,preferred,auto-right,1
monitor=,preferred,auto,auto,mirror,eDP-1


# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# Set programs that you use
$terminal = $HOME/.local/bin/kitty
$fileManager = nautilus
$menu = wofi --show drun
$menuAlt = wofi --show run

# Some default env vars.
env = XCURSOR_SIZE,24
env = QT_QPA_PLATFORMTHEME,qt6ct # change to qt6ct if you have that
env = ELECTRON_OZONE_PLATFORM_HINT,auto
env = MOZ_DISABLE_RDD_SANDBOX,1

env = XDG_SESSION_TYPE,wayland
# env = LIBVA_DRIVER_NAME,nvidia
# env = __GLX_VENDOR_LIBRARY_NAME,nvidia
# env = GBM_BACKEND,nvidia-drm

# env = NVD_BACKEND,direct

exec-once = hyprpm reload -n

# debug {
#     # overlay = true
#     disable_logs = false
#     enable_stdout_logs = true
# }

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = us
    kb_variant =
    kb_model =
    kb_options = compose:ralt
    kb_rules =

    follow_mouse = 2

    touchpad {
        natural_scroll = yes
    }

    # sensitivity = -0.1 # -1.0 - 1.0, 0 means no modification.  WIRELESS
    sensitivity = -0.8 # -1.0 - 1.0, 0 means no modification.
}
device {
    name = ydotoold-virtual-device
    kb_layout = us
    kb_variant =
    kb_options =
}
device {
    name = elan0303:00-04f3:31ff-touchpad
    sensitivity = -0.2
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 5
    gaps_out = 10
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    # layout = dwindle
    # layout = master
    layout = scroller

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10

    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    # enabled = no

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 2, myBezier
    animation = layers, 1, 0.5, default
    animation = windowsIn, 1, 2, default, popin 80%
    animation = windowsOut, 1, 2, default, popin 80%
    animation = border, 1, 3, default
    animation = borderangle, 1, 3, default
    animation = fade, 1, 3, default
    animation = workspaces, 1, 3, default
}

xwayland {
    force_zero_scaling = true
}

cursor {
    no_hardware_cursors = true
    hide_on_touch = true
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_status = master
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = on
}

misc {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    force_default_wallpaper = -1 # Set to 0 to disable the anime mascot wallpapers
}

plugin {
    hyprexpo {
        columns = 3
        gap_size = 5
        bg_col = rgb(111111)
        workspace_method = first current # [center/first] [workspace] e.g. first 1 or center m+1

        enable_gesture = true # laptop touchpad
        gesture_fingers = 3  # 3 or 4
        gesture_distance = 300 # how far is the "max"
        gesture_positive = false # positive = swipe down. Negative = swipe up.
    }

    scroller {
        column_default_width = twothirds
        focus_wrap = false
    }

    hyprbars {
        # example config
        bar_height = 20

        # example buttons (R -> L)
        # hyprbars-button = color, size, on-click
        hyprbars-button = rgb(ff4040), 10, 󰖭, hyprctl dispatch killactive
        hyprbars-button = rgb(eeee11), 10, , hyprctl dispatch fullscreen 1
    }
}


# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this.
# windowrulev2 = maximize,class:^(firefox)$
windowrulev2 = float, class:^(firefox)$,title:^(Picture-in-Picture)$
windowrulev2 = pin, class:^(firefox)$,title:^(Picture-in-Picture)$
windowrulev2 = size 800 450,class:^(firefox)$,title:^(Picture-in-Picture)$
windowrulev2 = monitor 2,class:^(firefox)$,title:^(Picture-in-Picture)$
windowrulev2 = move onscreen 100%-800 100%-450,class:^(firefox)$,title:^(Picture-in-Picture)$
windowrulev2 = move 0 0,title:^(flameshot)
windowrulev2 = suppressevent fullscreen,title:^(flameshot)
windowrulev2 = float, title:^(efck-chat-keyboard)
windowrulev2 = float, class:^(org.gnome.Calculator)$
windowrulev2 = float, class:^(hyprland-share-picker)$
windowrulev2 = float, class:^(org.keepassxc.KeePassXC)$
windowrulev2 = float, class:^(mcontrolcenter)$
windowrulev2 = float, class:^(com.nextcloud.desktopclient.nextcloud)$
windowrulev2 = float, class:^(ui-PlanAhea)$
windowrulev2 = move onscreen 100% 0, class:^(com.nextcloud.desktopclient.nextcloud)$
windowrulev2 = float, title:^(Calendar Reminders)$
windowrulev2 = float, class:^(com.github.hluk.copyq)$
windowrulev2 = float, class:^(zenity)$
windowrulev2 = nomaxsize, class:^(steam)$,title:^(Special Offers)$
windowrulev2 = float, class:^(Windscribe)$
windowrulev2 = nomaxsize, class:^(Windscribe)$
windowrulev2 = noshadow, class:^(Windscribe)$
windowrulev2 = noblur, class:^(Windscribe)$
windowrulev2 = noborder, class:^(Windscribe)$
windowrulev2 = opacity 0.0 override 0.0 override,initialClass:^(xwaylandvideobridge)$
windowrulev2 = noanim,initialClass:^(xwaylandvideobridge)$
windowrulev2 = noinitialfocus,initialClass:^(xwaylandvideobridge)$
windowrulev2 = maxsize 1 1,initialClass:^(xwaylandvideobridge)$
windowrulev2 = noblur,initialClass:^(xwaylandvideobridge)$
# windowrulev2 = maximize,class:^(VirtualBox Machine)$
# windowrulev2 = maximize,class:^(org.remmina.Remmina)$

workspace = 10,monitor:2


# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, return, exec, $terminal
bind = CTRL ALT, T, exec, $terminal
bind = $mainMod SHIFT, C, killactive, 
bind = $mainMod SHIFT, M, exit, 
bind = $mainMod, E, exec, $fileManager
bind = $mainMod, F, togglefloating, 
bind = $mainMod SHIFT, F, fullscreen,
bind = $mainMod, R, exec, $menu
bind = $mainMod SHIFT, R, exec, $menuAlt
bind = $mainMod, T, togglegroup
bind = , print, exec, grim -g "`slurp`"
bind = $mainMod, space, exec, efck-chat-keyboard
bind = CTRL, escape, exec, swaync-client -t -sw

# Laptop multimedia keys for volume and LCD brightness
bindel = ,XF86AudioRaiseVolume, exec, wpctl set-volume -l 1.5 @DEFAULT_AUDIO_SINK@ 5%+
bindel = ,XF86AudioLowerVolume, exec, wpctl set-volume -l 0 @DEFAULT_AUDIO_SINK@ 5%-
bindel = ,XF86AudioMute, exec, wpctl set-mute @DEFAULT_AUDIO_SINK@ toggle
bindel = ,XF86AudioMicMute, exec, wpctl set-mute @DEFAULT_AUDIO_SOURCE@ toggle
bindel = ,XF86MonBrightnessUp, exec, brightnessctl s 10%+
bindel = ,XF86MonBrightnessDown, exec, brightnessctl s 10%-

# Move focus with mainMod + arrow keys
bind = $mainMod CTRL, left, changegroupactive, b
bind = $mainMod CTRL, right, changegroupactive, f

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Move current workspace
bind = $mainMod, tab, movecurrentworkspacetomonitor, r
bind = $mainMod SHIFT, tab, movecurrentworkspacetomonitor, l

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# will switch to a submap called resize
bind = ALT, R, submap, resize

# will start a submap called "resize"
submap = resize
binde = , right, resizeactive, 10 0
binde = , left, resizeactive, -10 0
binde = , up, resizeactive, 0 -10
binde = , down, resizeactive, 0 10
bind = , escape, submap, reset
bind = , return, submap, reset
submap = reset

bind = $mainMod, P, submap, present
submap = present
binde = , m, exec, wl-present mirror
binde = , m, submap, reset
binde = , o, exec, wl-present set-output
binde = , r, exec, wl-present set-region
binde = Shift, r, exec, wl-present unset-region
binde = , c, exec, wl-present custom
bind = , escape, submap, reset
bind = , return, submap, reset
submap = reset


# pass through
# bind = SHIFT CONTROL,M,pass,^(WebCord)$
# bind = SHIFT CONTROL,M,pass,class:^(discord)$
# bind = SHIFT CONTROL,M,pass,^(wev)$
bind = SHIFTCONTROL,m,pass,^(vesktop)$
bind = SHIFTCONTROL,x,pass,title:^(Event Tester)$
bind = SHIFTCONTROL,x,pass,^(wev)$

# source = ~/.config/hypr/hyprland-master.conf
source = ~/.config/hypr/hyprland-scroller.conf



# exec = ${HOME}/.config/hypr/start-way-displays.sh
exec-once = hypridle
exec-once = /usr/lib/polkit-1/polkit-agent-helper-1 &
# exec-once = dunst &
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = systemctl --user start xdg-desktop-portal-gtk xdg-desktop-portal-hyprland.service

# auto start
exec-once = ${HOME}/.config/hypr/applets.sh
exec-once = ${HOME}/.config/hypr/autostart.sh


Config File: /home/constantin/.config/hypr/hyprland-scroller.conf: Read Succeeded

# {{{ Scroller

# Move focus with mainMod + arrow keys
bind = $mainMod, left, scroller:movefocus, l
# bind = $mainMod, left, scroller:alignwindow, c
bind = $mainMod, right, scroller:movefocus, r
# bind = $mainMod, right, scroller:alignwindow, c
bind = $mainMod, up, scroller:movefocus, u
# bind = $mainMod, up, scroller:alignwindow, c
bind = $mainMod, down, scroller:movefocus, d
# bind = $mainMod, down, scroller:alignwindow, c
bind = $mainMod, home, scroller:movefocus, begin
bind = $mainMod, end, scroller:movefocus, end

# Movement
bind = $mainMod CTRL, left, scroller:movewindow, l
bind = $mainMod CTRL, right, scroller:movewindow, r
bind = $mainMod CTRL, up, scroller:movewindow, u
bind = $mainMod CTRL, down, scroller:movewindow, d
bind = $mainMod CTRL, home, scroller:movewindow, begin
bind = $mainMod CTRL, end, scroller:movewindow, end

# Modes
bind = $mainMod, bracketleft, scroller:setmode, row
bind = $mainMod, bracketright, scroller:setmode, col

# Sizing keys
bind = $mainMod, equal, scroller:cyclesize, next
bind = $mainMod, minus, scroller:cyclesize, prev

# Admit/Expel
bind = $mainMod, I, scroller:admitwindow,
bind = $mainMod, O, scroller:expelwindow,

# Center submap
# will switch to a submap called center
bind = $mainMod, C, submap, center
# will start a submap called "center"
submap = center
# sets repeatable binds for resizing the active window
bind = , C, scroller:alignwindow, c
bind = , C, submap, reset
bind = , right, scroller:alignwindow, r
bind = , right, submap, reset
bind = , left, scroller:alignwindow, l
bind = , left, submap, reset
bind = , up, scroller:alignwindow, u
bind = , up, submap, reset
bind = , down, scroller:alignwindow, d
bind = , down, submap, reset
# use reset to go back to the global submap
bind = , escape, submap, reset
bind = , escape, submap, reset
# will reset the submap, meaning end the current one and return to the global one
submap = reset

# Fit size submap
# will switch to a submap called fitsize
bind = $mainMod, W, submap, fitsize
# will start a submap called "fitsize"
submap = fitsize
# sets binds for fitting columns/windows in the screen
bind = , W, scroller:fitsize, visible
bind = , W, submap, reset
bind = , right, scroller:fitsize, toend
bind = , right, submap, reset
bind = , left, scroller:fitsize, tobeg
bind = , left, submap, reset
bind = , up, scroller:fitsize, active
bind = , up, submap, reset
bind = , down, scroller:fitsize, all
bind = , down, submap, reset
# use reset to go back to the global submap
bind = , escape, submap, reset
bind = , escape, submap, reset
# will reset the submap, meaning end the current one and return to the global one
submap = reset

# overview keys
# bind key to toggle overview (normal)
bind = $mainMod, grave, scroller:toggleoverview
# overview submap
# will switch to a submap called overview
bind = $mainMod, grave, submap, overview
# will start a submap called "overview"
submap = overview
bind = , right, scroller:movefocus, right
bind = , left, scroller:movefocus, left
bind = , up, scroller:movefocus, up
bind = , down, scroller:movefocus, down
# use reset to go back to the global submap
bind = , escape, scroller:toggleoverview,
bind = , escape, submap, reset
bind = , return, scroller:toggleoverview,
bind = , return, submap, reset
bind = $mainMod, grave, scroller:toggleoverview,
bind = $mainMod, grave, submap, reset
# will reset the submap, meaning end the current one and return to the global one
submap = reset

# Marks
bind = $mainMod, M, submap, marksadd
submap = marksadd
bind = , a, scroller:marksadd, a
bind = , a, submap, reset
bind = , b, scroller:marksadd, b
bind = , b, submap, reset
bind = , c, scroller:marksadd, c
bind = , c, submap, reset
bind = , escape, submap, reset
bind = , escape, submap, reset
submap = reset

bind = $mainMod SHIFT, M, submap, marksdelete
submap = marksdelete
bind = , a, scroller:marksdelete, a
bind = , a, submap, reset
bind = , b, scroller:marksdelete, b
bind = , b, submap, reset
bind = , c, scroller:marksdelete, c
bind = , c, submap, reset
bind = , escape, submap, reset
bind = , escape, submap, reset
submap = reset

bind = $mainMod, apostrophe, submap, marksvisit
submap = marksvisit
bind = , a, scroller:marksvisit, a
bind = , a, submap, reset
bind = , b, scroller:marksvisit, b
bind = , b, submap, reset
bind = , c, scroller:marksvisit, c
bind = , c, submap, reset
bind = , escape, submap, reset
bind = , escape, submap, reset
submap = reset

bind = $mainMod CTRL, M, scroller:marksreset

# }}}

======Config-End========
```

</details>

### Description

Surface is drawn in small part of window, screenshot doesn't work

### How to reproduce

Launch an electron app (e.g. draw.io or vesktop) with `--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations`

Started after 0c7a7e2d, currently on bc299928, will try to do a bisect shortly

**Comment by cpiber** [comment]
![Screenshot 2024-10-09 17-01-21](https://github.com/user-attachments/assets/773855c5-accb-47eb-b0b8-70e88a02e0a2)

Not sure why `grim` doesn't work anymore either, could be related or not, here is one from OBS

**Comment by vaxerski** [comment]
can you try latest git

**Comment by cpiber** [comment]
Git bisect lead me to bc299928 and I had to undo the change in `updateState`, which I see has been fixed in 223dcc8bac7fee90aa6aa7596d9379e56b526e62
